ex_functions/csvParser.js

```diff
@@ -0,0 +1,48 @@
+const file=`Name;Street;City;Age
+Peter Pan;Am Hang 5;12345 Einsam;42
+Maria Schmitz;Kölner Straße 45;50123 Köln;43
+Paul Meier;Münchener Weg 1;87654 München;65`;
+
+function csvParser(file, separator, spaceBefore, spaceAfter){
```

Comment:
1 problema importante coa estratexia: estás lendo a entrada e a saida ao mesmo tempo e esto implica:
- mestura de interfaz de entrada e saída (noutros casos sería interfaz e persistencia)
- máis difícil modificas (evolucionar) ambas partes por separado
- estás recorrendo as columnas 2 veces, un pouco problema de eficiencia en entradas de moitas liñas
- creo que estas decidindo o ancho das columnas só polo cabeceira

A miña recomendación sería partir a solución en 2:
- primeira parte, saco os datos das entradas e calculas anchos
- segunda parte, uso o procesado para xerar a saída